src/ray/rpc/worker/core_worker_client_pool.h

```diff
@@ -72,8 +67,9 @@ class CoreWorkerClientPool {
 
   /// A pool of open connections by WorkerID. Clients can reuse the connection
   /// objects in this pool by requesting them.
-  absl::flat_hash_map<ray::WorkerID, shared_ptr<CoreWorkerClientInterface>> client_map_
+  absl::flat_hash_map<ray::WorkerID, std::weak_ptr<CoreWorkerClientInterface>> client_map_
```

Comment:
Isn't this likely to regress worker->worker performance?

Reply:
@ericl  check [this](https://github.com/ray-project/ray/pull/33087/files#r1127283425) comment for explanation 

We'll run related nightly test too to prevent the regression.